src/learning-catalogue/model/factory/courseFactory.ts

```diff
@@ -0,0 +1,38 @@
+import {Course} from '../course'
+import {ModuleFactory} from './moduleFactory'
+import {EventFactory} from './eventFactory'
+import {AudienceFactory} from './audienceFactory'
+
+export class CourseFactory {
+	private _moduleFactory: ModuleFactory
+
+	constructor() {
+		this._moduleFactory = new ModuleFactory(
+			new AudienceFactory(),
+			new EventFactory()
+		)
```

Comment:
Not using dependency injection here? 

Also, if CourseFactory can create it's own ModuleFactory shouldn't ModuleFactory create it's own AudienceFactory and EventFactory?

Reply:
Fair point I wasn't very happy with this. I started using dependency injection here but thought it exposed internal classes to users of the API. I decided to have defaults that could be overridden if necessary (like in testing). The ModuleFactory doesn't have setters for AudienceFactory and EventFactory as they can be passed in the constructor so weren't necessary for testing.